### ISSUE TITLE
No Data to Export Should Inform User / Not Generate Report

### ISSUE THREAD

**Issue by davidclaveau** [bcgov/bcparks-ar-admin#384]
#### Description:

When I choose to generate a report, if there are no values found then an empty report should not be generated and I should be informed on the frontend that the report is empty and not needed.

The user doesn't see that a report is empty unless they download the report and open the spreadsheet (and find there is no data). This could be applied to Variance or Missing.

#### Acceptance Criteria:

Given I am an A&R user
When I choose to generate a report
Then when there's no data, the UI tells me and does not create an empty report (and it prevents me from downloading an empty report).

#### Development Checklist:

- [ ] User can still attempt to generate a report
- [ ] When there's no data to generate a report, the user is told on the frontend
- [ ] No report is generated on the backend

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by davidclaveau** [comment]
Hey @manuji this is ready to be moved to TEST now!

This is just that nice-to-have change that you had mentioned, which Nicole and Sally had agreed would be good to add. If there's no data to export the backend won't create a document/upload it to S3, and the frontend will now make note for the user and show 'No data - no report generated'.

**Note that it's currently just for the Missing Export** as I didn't want to step on the toes of @meredithom's changes coming in #254, but it should be pretty easy to add once that's done 😄 

![Image](https://github.com/user-attachments/assets/e4542ab3-bf90-47b7-88ba-af39f7122001)

**Comment by Dianadec** [comment]
@davidclaveau is this one ready to be tested or is it waiting on a piece from #254 ? cc @manuji

**Comment by manuji** [comment]
Hi @Dianadec , this is ready for QA and I will update the status of this today as well. My apologies for the delay.

**Comment by manuji** [comment]
Tested on TEST: **Passed**
* Verified that this task has been implemented and working as expected. 
* Passing to PO review.

**Comment by LindsayMacfarlane** [comment]
Tested on TEST: **Passed**

- Verified on TEST by going to export report, selecting FY 2019-2020 and searching for Strathcona Park. 
- Exporter status: 'No data - no report generated'.

Thanks so much @davidclaveau !
